lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1556,6 +1545,21 @@ def _create_node_arm_parameters(
                         plan_publisher=plan_publisher,
                     )
 
+        # Set Hyper-V Generation
+        # Note: Image capability has already been merged with the VM capability
+        hyperv_generation = capability._find_feature_by_type(
+            features.HyperVGenerationSettings.type, capability.features
+        )
+        assert hyperv_generation, "Node space must have Hyper-V Generation defined."
```

Comment:
Does it need a `if hyperv_generation:` block? Is it possible `_find_feature_by_type` returns None?